src/SurfaceFluxes.jl

```diff
@@ -65,156 +77,112 @@ end
     println(io, "L_MO           = ", sfc.L_MO)
-    println(io, "wθ_flux_star = ", sfc.wθ_flux_star)
     println(io, "flux           = ", sfc.flux)
     println(io, "x_star         = ", sfc.x_star)
     println(io, "C_exchange     = ", sfc.C_exchange)
     println(io, "-----------------------")
 end
 
-function surface_fluxes_f!(F, x, nt)
-    param_set = nt.param_set
-    wθ_flux_star = nt.wθ_flux_star
-    z_in = nt.z_in
-    z_0 = length(nt.z_0) > 1 ? Tuple(nt.z_0) : nt.z_0
-    x_in = Tuple(nt.x_in)
-    x_s = Tuple(nt.x_s)
-    n_vars = nt.n_vars
-    scheme = nt.scheme
-    universal_func = nt.universal_func
-    θ_scale = nt.θ_scale
-
-    x_tup = Tuple(x)
-
-    u_star, θ_star = x_tup[2], x_tup[3]
-    if wθ_flux_star == nothing
-        wθ_surf_flux = -u_star * θ_star
-    else
-        wθ_surf_flux = wθ_flux_star
-    end
-    L_MO = monin_obukhov_length(param_set, u_star, θ_scale, wθ_surf_flux)
-    uf = universal_func(param_set, L_MO)
-    F_nt = ntuple(Val(n_vars + 1)) do i
-        if i == 1
-            F_i =
-                x_tup[1] - monin_obukhov_length(
-                    param_set,
-                    u_star,
-                    θ_scale,
-                    wθ_surf_flux,
-                )
-        else
-            ϕ = x_tup[i]
-            transport = i - 1 == 1 ? UF.MomentumTransport() : UF.HeatTransport()
-            F_i =
-                ϕ - compute_physical_scale(
-                    uf,
-                    z_in,
-                    length(nt.z_0) > 1 ? z_0[i - 1] : z_0,
-                    x_in[i - 1],
-                    x_s[i - 1],
-                    transport,
-                    scheme,
-                )
-        end
-        F_i
-    end
-    F .= F_nt
-end
-
 """
-    surface_conditions
-Surface conditions given
- - `MO_param_guess` initial guess for solution (`L_MO, u_star, θ_star, ϕ_star, ...`)
- - `x_in` Inner values for state variable array `x`. They correspond to volume averages in FV and
-    nodal point values in DG
- - `x_s` surface values for state variable array `x`
- - `z_0` roughness lengths for state variable array `x`
- - `θ_scale` virtual dry static energy (i.e., basic potential temperature)
- - `z_in` Input height for the similarity functions. It is Δz for FV and the height
-    of the first nodal point for DG
- - `wθ_flux_star` potential temperature flux (optional)
-  - `universal_func` family of flux-gradient universal functions used (optional)
+  Return Surface Conditions based on combination of inputs 
+  Cases are 
+  1) 
+  2)
+  3)
+  4)
 
-If `wθ_flux_star` is not given, then it is computed by iteration
-of equations 3, 17, and 18 in Nishizawa2018.
 """
 function surface_conditions(
     param_set::APS,
-    MO_param_guess::AbstractVector,
-    x_in::AbstractVector,
-    x_s::AbstractVector,
+    L_MO_init::FT,
+    ts_in::TD.ThermodynamicState,
+    ts_sfc::TD.ThermodynamicState,
+    u_in::FT, u_sfc::FT,
     z_0::Union{AbstractVector, FT},
-    θ_scale::FT,
```

Comment:
Why is `z_0` above still a vector? Can this be changed to a float?

Reply:
z_0 might be a list of value for momentum (z_0m) for heat and moisture (z_0h), (z_0q) ...